### PR TITLE
Integration test suite: separate kompilation from execution

### LINKED ISSUE #2954: Integration test suite: separate kompilation from execution
We should remove the `VERIFICATION` module where it's not needed. This means we need different ways of running tests: the specs which depend on such a module will be run as they are now (by re-kompiling the definition for each spec), and those which don't should just be run simply.

### PR DESCRIPTION
**Pull request by andreiburdusa**
Fixes #2954 

See https://github.com/kframework/kore/issues/2954#issuecomment-1028781943

### Scope:
- Avoid unnecessary kompilation
- Simplify testing infrastructure so we can run the integration tests twice in https://github.com/kframework/kore/pull/2953

### Estimate:
- February 7

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by andreiburdusa** [comment]
These changes allow us to run
```
$MAKE kore
$MAKE build-test
$MAKE test-k
$MAKE clean-execution
$MAKE test-k-simplifierx
```
in the `integration-.sh` script.
This way the tests can be ran twice (with and without `--simplifierx`) without kompiling the definitions twice.

File: test/imp/imp.k [hunk 1]
```diff
@@ -206,4 +206,34 @@ a computational step. */
...
+            | "sum"   [token]
+
+  // for sum-save-proofs-spec
+  syntax Id ::= "n"     [token]
+            | "sum"   [token]
+
+  //for unreachable-spec
+  syntax Id ::= "a"  [token]
+            | "b"  [token]
+            | "r"  [token]
```
Comment (ana-pantilie) [inline test/imp/imp.k:237]: Some of these tokens are duplicated, that can't be good, right?

File: test/include.mk [hunk 2]
```diff
@@ -104,13 +104,13 @@ PATTERN_OPTS = --pattern "$$(cat $*.k)"
 	@echo "KORE_EXEC_OPTS =" $(KORE_EXEC_OPTS)
 	rm -f $@
 	$(if $(STORE_PROOFS),rm -f $(STORE_PROOFS),$(if $(RECALL_PROOFS),cp $(RECALL_PROOFS) $(@:.out=.save-proofs.kore)))
-	$(KOMPILE) $(KOMPILE_OPTS) --main-module $(KPROVE_MODULE) $(KPROVE_SPEC)
-	rm -rf $*-tmpdir; mkdir $*-tmpdir
-	mv $*-spec-kompiled $*-tmpdir
-	$(KPROVE) $(KPROVE_OPTS) -d $*-tmpdir $(KPROVE_SPEC_OPTS) $(KPROVE_SPEC) >$@ || true
+#	$(KOMPILE) $(KOMPILE_OPTS) --main-module $(KPROVE_MODULE) $(KPROVE_SPEC)
+#	rm -rf $*-tmpdir; mkdir $*-tmpdir
+#	mv $*-spec-kompiled $*-tmpdir
```
Comment (ana-pantilie) [inline test/include.mk:109]: Did you forget to remove the commented out lines here?

File: test/include.mk [hunk 3]
```diff
@@ -104,13 +104,13 @@ PATTERN_OPTS = --pattern "$$(cat $*.k)"
...
+	$(KPROVE) $(KPROVE_OPTS) $(KPROVE_SPEC_OPTS) $(KPROVE_SPEC) >$@ || true
 	$(DIFF) $@.golden $@ || $(FAILED)
 	$(if $(STORE_PROOFS),$(DIFF) $(STORE_PROOFS).golden $(STORE_PROOFS) || $(FAILED_STORE_PROOFS))
-	rm -rf $*-tmpdir
+#	rm -rf $*-tmpdir
```
Comment (ana-pantilie) [inline test/include.mk:113]: This as well?

File: test/map-symbolic/remove-9-spec.k [hunk 4]
```diff
@@ -3,12 +3,8 @@
 
 requires "map-tests.k"
 
-module VERIFICATION
-    imports MAP-TESTS
-endmodule
-
 module REMOVE-9-SPEC
-    imports VERIFICATION
+    imports MAP-TESTS
```
Comment (ana-pantilie) [inline test/map-symbolic/remove-9-spec.k:7]: Did you modify https://github.com/kframework/kore/blob/3eefcb078b24128e5773af9dd1170a19230dd832/test/map-symbolic/gen-tests.sh?

For these tests in map-symbolic, you must modify that generator script, and generate them with the script.

File: test/no-injection-into-hooked/Makefile [hunk 5]
```diff
@@ -23,7 +25,9 @@ golden:
 clean:
 	rm -fr $(KOMPILED)
 
-.PHONY: test-k test golden clean
+clean-execution:
```
Comment (ana-pantilie) [inline test/no-injection-into-hooked/Makefile:28]: What does this target do?

File: test/parser/Makefile [hunk 6]
```diff
@@ -2,6 +2,7 @@ ifeq ($(origin TOP), undefined)
 	TOP = $(shell git rev-parse --show-toplevel)
 endif
 
+DEF_KORE =
```
Comment (ana-pantilie) [inline test/parser/Makefile:5]: Why did you add this variable?

File: test/regression-evm/Makefile [hunk 7]
```diff
@@ -1,3 +1,4 @@
+DEF_KORE =
```
Comment (ana-pantilie) [inline test/regression-evm/Makefile:1]: Same question as above.

File: test/regression-wasm/Makefile [hunk 8]
```diff
@@ -1,3 +1,4 @@
+DEF_KORE =
```
Comment (ana-pantilie) [inline test/regression-wasm/Makefile:1]: Same question as above.

File: test/set/set-iteration/test.k [hunk 9]
```diff
@@ -18,4 +18,18 @@ module TEST
...
+                    | doneTest ( StepSort )
+  // ------------------------------------------
+  rule runTest( T ) => doneTest( T )
+
+  // for test-2-spec
+  syntax StepSort ::= Set
+  syntax    KItem ::= runTest  ( StepSort )
+                    | doneTest ( StepSort )
+ // ------------------------------------------
+  rule runTest( T ) => doneTest( T )
```
Comment (ana-pantilie) [inline test/set/set-iteration/test.k:33]: This is duplicated again.

File: test/smc/smc.k [hunk 10]
```diff
@@ -100,4 +100,39 @@ module SMC
...
+  syntax Id ::= "max" [token]
+
+  // for sum-spec
+  syntax Id ::= "n"     [token]
+              | "sum"   [token]
+
+  // for sum-stack-spec
+  syntax Id ::= "n"     [token]
+              | "sum"   [token]
+
```
Comment (ana-pantilie) [inline test/smc/smc.k:137]: Watch out for the duplication here as well.

**Review by ana-pantilie — CHANGES_REQUESTED** [review]
Some questions and some requests.

File: test/map-symbolic/remove-9-spec.k (hunk 4 above)
Comment (andreiburdusa) [reply test/map-symbolic/remove-9-spec.k:7]: How should that script be ran? `./gen-tests.sh` doesn't have effect

Comment (andreiburdusa) [reply test/map-symbolic/remove-9-spec.k:7]: How should that script be ran? ./gen-tests.sh doesn't have effect

File: test/no-injection-into-hooked/Makefile (hunk 5 above)
Comment (andreiburdusa) [reply test/no-injection-into-hooked/Makefile:28]: After running `make test-k`, running `make test-k-simplifierx` printed "nothing to be done". So I added `clean-execution:` in order to remove the files created during execution, and not the kompiled directory.

File: test/parser/Makefile (hunk 6 above)
Comment (andreiburdusa) [reply test/parser/Makefile:5]: Oh, actually it looks like it's only needed in `regression-evm` and `regression-wasm`

File: test/regression-evm/Makefile (hunk 7 above)
Comment (andreiburdusa) [reply test/regression-evm/Makefile:1]: `make build-test` is called for each test, and if I don't set `DEF_KORE` to be nothing, it's going to automatically search for a `.k` definition, which in this case doesn't exist.
I added the target `build-test` because Everett suggested it, but now I'm not so sure it's needed for what we are trying to do here.

File: test/imp/imp.k (hunk 1 above)
Comment (andreiburdusa) [reply test/imp/imp.k:237]: I'll take care of all duplication

File: test/include.mk (hunk 2 above)
Comment (andreiburdusa) [reply test/include.mk:109]: Yes

**Review by ana-pantilie — APPROVED** [review]
